src/main/java/application/FormController.java

```diff
@@ -336,8 +336,30 @@ public FormEntryNavigationResponseBean getNext(@RequestBody SessionRequestBean r
         return responseBean;
     }
 
+    @ApiOperation(value = "Get the questions for the next index for SMS")
+    @RequestMapping(value = Constants.URL_NEXT, method = RequestMethod.POST)
+    @ResponseBody
+    @UserLock
+    @UserRestore
+    public FormEntryNavigationResponseBean getNextSms(@RequestBody SessionRequestBean requestBean,
+                                                   @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken) throws Exception {
+        SerializableFormSession serializableFormSession = formSessionRepo.findOneWrapped(requestBean.getSessionId());
+        storageFactory.configure(serializableFormSession.getUsername(),
+                serializableFormSession.getDomain(),
+                serializableFormSession.getAppId(),
+                serializableFormSession.getAsUser()
+        );
```

Comment:
confirming that all this works when it's a case filling out the survey as well as when it's a user filling it out?